From 5f05454b30da807a02233e99373a37b7c863d5c3 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 26 Jul 2018 17:13:32 +0500
Subject: [PATCH] configure.ac: adjust usage of LLVM flags

The llvm-config is provided through a native build
so using flags from there creates conflicts between
mesa and llvm on the target.

Signed-off-by: Awais Belal <awais_belal@mentor.com>

---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index f59b29f2d68..893d41f7094 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1101,6 +1101,7 @@ strip_unwanted_llvm_flags() {
         -e 's/[[[:space:]]]+-pedantic[[[:space:]]]/ /g' \
         -e 's/[[[:space:]]]+-W[[^[:space:]]]*//g' \
         -e 's/[[[:space:]]]+-O[[^[:space:]]]*//g' \
+        -e 's/[[[:space:]]]+-D_GLIBCXX_USE_CXX11_ABI[[^[:space:]]]*//g' \
         -e 's/[[[:space:]]]+-g[[^[:space:]]]*//g' \
         -e 's/-fno-rtti[[[:space:]]]/-Fno-rtti /g' \
         -e 's/[[[:space:]]]+-f[[^[:space:]]]*//g' \
@@ -2900,7 +2901,7 @@ dnl
 if test "x$enable_llvm" = xyes; then
     DEFINES="${DEFINES} -DHAVE_LLVM=0x0$LLVM_VERSION_INT -DMESA_LLVM_VERSION_PATCH=$LLVM_VERSION_PATCH"
 
-    LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
+    LLVM_LDFLAGS=
     LLVM_CFLAGS=$LLVM_CPPFLAGS   # CPPFLAGS seem to be sufficient
     LLVM_CXXFLAGS=`strip_unwanted_llvm_flags "$LLVM_CONFIG --cxxflags"`
     LLVM_CXXFLAGS="$CXX11_CXXFLAGS $LLVM_CXXFLAGS"
